Look for RedisClient in all modules

glefloch opened the pull request:
This branch looks for redis client in all archives. I tested it in a multi module project locally. Is there any place where I could put such a test? the `integration-test/redis-client` is not a multi-module project. 

close #15581

glefloch committed:
Look for RedisClient in all modules

mkouba commented:
There's a formatting error:
```
 File '/home/runner/work/quarkus/quarkus/extensions/redis-client/deployment/src/main/java/io/quarkus/redis/client/deployment/RedisClientProcessor.java' has not been previously formatted.  Please format file and commit before running validation!
```

mkouba commented on extensions/redis-client/deployment/src/main/java/io/quarkus/redis/client/deployment/RedisClientProcessor.java:
```diff
@@ -86,10 +87,12 @@ public void produceRedisClient(RedisClientRecorder recorder, ApplicationArchives
-        Collection<AnnotationInstance> clientAnnotations = indexView.getAnnotations(REDIS_CLIENT_ANNOTATION);
-        for (AnnotationInstance annotation : clientAnnotations) {
-            clientNames.add(annotation.value().asString());
+        for (ApplicationArchive applicationArchive: applicationArchives.getAllApplicationArchives()) {
```
You can also leverage the `io.quarkus.arc.deployment.BeanArchiveIndexBuildItem`.

machi1990 replied:
The same will have to be done for the mongo-extension https://github.com/quarkusio/quarkus/blob/33364c8700988293515dfb34d53f238f196ead15/extensions/mongodb-client/deployment/src/main/java/io/quarkus/mongodb/deployment/MongoClientProcessor.java#L119?

glefloch replied:
@mkouba Yes, I will use the `BeanArchiveIndexBuildItem` this will remove a for-loop

glefloch commented:
I fixed the formatting error, I pushed to fast yesterday :)

glefloch commented:
@mkouba I updated the code with `BeanArchiveIndexBuildItem` and fixed the formatted. 

@machi1990 for the mongo extension, should we do  that in an other branch ?
